### PR TITLE
Add catalog to dependencies to update by Renovate

### PR DESCRIPTION
**Pull request by krysal**
## Description

I noticed the Catalog was missing from these updates, but also why is it limited to development dependencies?

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by AetherUnbound — APPROVED** [review]
This seems okay!

> I noticed the Catalog was missing from these updates, but also why is it limited to development dependencies?

When you say "it", what are you referring to? Many of the Airflow dependencies are limited by the constraints Airflow provides, so we try not to pin those dependencies directly.

**Comment by krysal** [comment]
@AetherUnbound I'm refering to that group of dependencies, it includes all the python projects, not only the catalog so it looked strange to me.

https://github.com/WordPress/openverse/blob/9b44496703708454198cb80dc18a630f63ed64e6/.github/renovate.json#L67-L70

**Review by dhruvkb — COMMENTED** [review]
Isn't the catalog covered by lines 58-61?

https://github.com/WordPress/openverse/pull/3465/files#diff-f82f7d36a61d22f640c25bdb8b0435bf9cb38679d99d5f460753fa668ee7cdb3R58-R61

**Comment by krysal** [comment]
@dhruvkb I believe that is only for putting the label on the files that are under the catalog folder 🤔

**Review by sarayourfriend — APPROVED** [review]
LGTM.

IIRC I excluded the catalog dependencies from that list because I wasn't sure if its development dependencies could get upgraded in a grouped upgrade due to Airflow's constraints on dependency management. I can't think of what the obstacle I imagined to exist was, but it should be fine.

The group modified in this PR is just to group development dependencies, though. Renovate will automatically include the catalogue's requirements.txt files when considering upgrades. It differs from dependabot in this way, you don't need to explicitly include each thing you want renovate to manage. Instead it defaults to managing updates for everything, and you only have to configure excludes, labeling rules, and dependency groups if you want grouped update PRs (instead of an individual PR per dependency).

In other words, it already handles the catalogue's dependencies, it just opens a PR for each one. This will make it so that its development dependencies join the others and get upgraded together in a single PR, but the non-development dependencies will still continue to get the one-PR-per-upgrade treatment that renovate should be giving them already.
